### PR TITLE
Add table oci_mysql_configuration. Closes #166

### LINKED ISSUE #166: Add table oci_mysql_configuration
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro oci-test % ./tint.js oci_mysql_configuration
Test names: [ 'tests/oci_mysql_configuration' ]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_mysql_configuration []

PRETEST: tests/oci_mysql_configuration

TEST: tests/oci_mysql_configuration
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.local_file.configuration will be read during apply
  # (config refers to values not yet known)
 <= data "local_file" "configuration"  {
      + content        = (known after apply)
      + content_base64 = (known after apply)
      + filename       = "/private/var/folders/df/957ty4t146vfy54wm4254kp40000gp/T/tests/oci_mysql_configuration/terraform/test/output.json"
      + id             = (known after apply)
    }

  # null_resource.named_test_resource will be created
  + resource "null_resource" "named_test_resource" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + lifecycle_state = (known after apply)
  + resource_id     = (known after apply)
  + resource_name   = (known after apply)
  + shape_name      = (known after apply)
  + tenancy_ocid    = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
  + type            = (known after apply)
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci mysql configuration list --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq --output json > /private/var/folders/df/957ty4t146vfy54wm4254kp40000gp/T/tests/oci_mysql_configuration/terraform/test/output.json"]
null_resource.named_test_resource (local-exec): WARNING: Permissions on /Users/arnab/.oci/config are too open.
null_resource.named_test_resource (local-exec): To fix this please try executing the following command:
null_resource.named_test_resource (local-exec): oci setup repair-file-permissions --file /Users/arnab/.oci/config
null_resource.named_test_resource (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.named_test_resource (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.named_test_resource (local-exec): WARNING: Permissions on /Users/arnab/PEM/-08-12-10-19.pem are too open.
null_resource.named_test_resource (local-exec): To fix this please try executing the following command:
null_resource.named_test_resource (local-exec): oci setup repair-file-permissions --file /Users/arnab/PEM/-08-12-10-19.pem
null_resource.named_test_resource (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.named_test_resource (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.named_test_resource: Creation complete after 3s [id=8640282432854554245]
data.local_file.configuration: Reading...
data.local_file.configuration: Read complete after 0s [id=19d512defc0f2b5a7dffacb589364fc8513d3c91]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

lifecycle_state = "ACTIVE"
resource_id = "ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgh"
resource_name = "BM.Standard.E2.64.Standalone"
shape_name = "BM.Standard.E2.64"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
type = "DEFAULT"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "BM.Standard.E2.64.Standalone",
    "id": "ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgh",
    "lifecycle_state": "ACTIVE",
    "shape_name": "BM.Standard.E2.64",
    "type": "DEFAULT"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "BM.Standard.E2.64.Standalone",
    "id": "ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgh",
    "lifecycle_state": "ACTIVE",
    "shape_name": "BM.Standard.E2.64",
    "type": "DEFAULT"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "BM.Standard.E2.64.Standalone"
  }
]
✔ PASSED

POSTTEST: tests/oci_mysql_configuration

TEARDOWN: tests/oci_mysql_configuration

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  display_name,
  id,
  description,
  lifecycle_state as state,
  time_created
from
  oci_mysql_configuration;
+-------------------------------------------+--------------------------------------------------------------------------------------------+------------------------------------------------------------------
| display_name                              | id                                                                                         | description                                                      
+-------------------------------------------+--------------------------------------------------------------------------------------------+------------------------------------------------------------------
| MySQL.VM.Standard.E3.48.768GB.Standalone  | ocid1.mysqlconfiguration.oc1..aaaaaaaaep75r2j453khbhuq7kfmxnefskx4wdw5he4r7rnr75fvkrtzkf6q | Default Standalone configuration for the MySQL.VM.Standard.E3.48.
| VM.Standard.E2.8.Standalone               | ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgl | Default Standalone configuration for the VM.Standard.E2.8 MySQL S
| MySQL.VM.Standard.E3.4.64GB.HA            | ocid1.mysqlconfiguration.oc1..aaaaaaaaroue27ea2a2aahfag3tun26akpicz6agzsdgqfncfh54sd3llcua | Default HA configuration for the MySQL.VM.Standard.E3.4.64GB MySQ
| MySQL.VM.Standard.E3.8.128GB.Standalone   | ocid1.mysqlconfiguration.oc1..aaaaaaaa6pxyzjrtsldmu34ynebrvf4ev535qqk4foxqrblnmfr6fym5bm2q | Default Standalone configuration for the MySQL.VM.Standard.E3.8.1
| MySQL.VM.Standard.E3.8.128GB.HA           | ocid1.mysqlconfiguration.oc1..aaaaaaaalhjbtt5tffdk64tb477gcpwlgbxse54qga7brjfuwpneleizutva | Default HA configuration for the MySQL.VM.Standard.E3.8.128GB MyS
| MySQL.VM.Standard.E3.16.256GB.Standalone  | ocid1.mysqlconfiguration.oc1..aaaaaaaafbsxowzfzxnnvm5mmmutrfl73hs2cv7ujgs7zqnzzw7v52fvihda | Default Standalone configuration for the MySQL.VM.Standard.E3.16.
| MySQL.HeatWave.BM.Standard.E3.Standalone  | ocid1.mysqlconfiguration.oc1..aaaaaaaap7thcvkv3za4bbesmugyhtv7ogns7yvxiaw65c3c2efn22qb3iwa | Default Standalone configuration for the MySQL.HeatWave.BM.Standa
| MySQL.VM.Standard.E3.16.256GB.HA          | ocid1.mysqlconfiguration.oc1..aaaaaaaafnjyqej2hkh7or2rhq2p3nvfrzlgjjeqsjqm7bbzs7svnarb75xa | Default HA configuration for the MySQL.VM.Standard.E3.16.256GB My
| MySQL.HeatWave.VM.Standard.E3.Standalone  | ocid1.mysqlconfiguration.oc1..aaaaaaaanivlpzq2ewropx574nrmoghupa5gy54yfdre26e664q7rwujpyxq | Default Standalone configuration for the MySQL.HeatWave.VM.Standa
| MySQL.VM.Standard.E3.1.8GB.HA             | ocid1.mysqlconfiguration.oc1..aaaaaaaantprksu6phqfgr5xvyut46wdfesdszonbclybfwvahgysfjbrb4q | Default HA configuration for the MySQL.VM.Standard.E3.1.8GB MySQL
| MySQL.VM.Standard.E3.64.1024GB.Standalone | ocid1.mysqlconfiguration.oc1..aaaaaaaalic45e2gauld2mwdpppzydow2sfkpcvecbfizkvgkhupezoi7oja | Default Standalone configuration for the MySQL.VM.Standard.E3.64.
| VM.Standard.E2.1.Standalone               | ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgi | Default Standalone configuration for the VM.Standard.E2.1 MySQL S
| MySQL.VM.Standard.E3.24.384GB.HA          | ocid1.mysqlconfiguration.oc1..aaaaaaaauez53axdqqx2whwvcqvdhxx7by36rlcxxg5hakdqh2jvh4dh4q2a | Default HA configuration for the MySQL.VM.Standard.E3.24.384GB My
| MySQL.VM.Standard.E3.24.384GB.Standalone  | ocid1.mysqlconfiguration.oc1..aaaaaaaajeya2xmvgcwjqsuevsmgsry4frwpnfxyfpggkuff2kbh2hh2ugxq | Default Standalone configuration for the MySQL.VM.Standard.E3.24.
| MySQL.VM.Standard.E3.1.16GB.HA            | ocid1.mysqlconfiguration.oc1..aaaaaaaahgiqob6oo3cwcoq2nbc3u2wz4uwlwblgyes6veogg6rv564aexwq | Default HA configuration for the MySQL.VM.Standard.E3.1.16GB MySQ
| VM.Standard.E2.4.Standalone               | ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgk | Default Standalone configuration for the VM.Standard.E2.4 MySQL S
| MySQL.VM.Standard.E3.1.16GB.Standalone    | ocid1.mysqlconfiguration.oc1..aaaaaaaaweky3hvms4bxw2uyf6pjcoxzf5gkqzxvxmzolgcu6bufd5bjnpyq | Default Standalone configuration for the MySQL.VM.Standard.E3.1.1
| MySQL.VM.Standard.E3.32.512GB.HA          | ocid1.mysqlconfiguration.oc1..aaaaaaaaegi6wwci2buv2duyoklhbulhmiztesp6eynqvruclupmaxo52d5a | Default HA configuration for the MySQL.VM.Standard.E3.32.512GB My
| MySQL.VM.Standard.E3.1.8GB.Standalone     | ocid1.mysqlconfiguration.oc1..aaaaaaaalwzc2a22xqm56fwjwfymixnulmbq3v77p5v4lcbb6qhkftxf2trq | Default Standalone configuration for the MySQL.VM.Standard.E3.1.8
| MySQL.VM.Standard.E3.2.32GB.HA            | ocid1.mysqlconfiguration.oc1..aaaaaaaah4ayiyeentcvebzlc5zw3u5otdlpupwkoyj566dukhp4ymz2yseq | Default HA configuration for the MySQL.VM.Standard.E3.2.32GB MySQ
| MySQL.VM.Standard.E3.32.512GB.Standalone  | ocid1.mysqlconfiguration.oc1..aaaaaaaacmprv4xraedqw3zk6ls6jweiold3gsbtfigxf4xujuwq3nxaqava | Default Standalone configuration for the MySQL.VM.Standard.E3.32.
| MySQL.VM.Standard.E3.2.32GB.Standalone    | ocid1.mysqlconfiguration.oc1..aaaaaaaakremacvh2fizcznnja5rdxry2q4nyn27afjblyrimzjmrqblhfwa | Default Standalone configuration for the MySQL.VM.Standard.E3.2.3
| BM.Standard.E2.64.Standalone              | ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgh | Default Standalone configuration for the BM.Standard.E2.64 MySQL 
| VM.Standard.E2.2.Standalone               | ocid1.mysqlconfiguration.oc1..aaaaaaaah6o6qu3gdbxnqg6aw56amnosmnaycusttaa7abyq2tdgpgubvsgj | Default Standalone configuration for the VM.Standard.E2.2 MySQL S
| MySQL.VM.Standard.E3.48.768GB.HA          | ocid1.mysqlconfiguration.oc1..aaaaaaaa6q6vod4vlo5pn7w3blmjxouv4dv3ke24zt4dhat7jqmwhh2bnltq | Default HA configuration for the MySQL.VM.Standard.E3.48.768GB My
| MySQL.VM.Standard.E3.64.1024GB.HA         | ocid1.mysqlconfiguration.oc1..aaaaaaaa2tiusjgxkqd344nbiv32rquejgzzo5mgwmvlpg2cay5ixl7sjv3a | Default HA configuration for the MySQL.VM.Standard.E3.64.1024GB M
| MySQL.VM.Standard.E3.4.64GB.Standalone    | ocid1.mysqlconfiguration.oc1..aaaaaaaam2bdlacmi4d2p33wx3aqqx5clnorpokoa6hxezz75rhw46xlj3ka | Default Standalone configuration for the MySQL.VM.Standard.E3.4.6
+-------------------------------------------+--------------------------------------------------------------------------------------------+------------------------------------------------------------------
> select
  display_name,
  id,
  description,
  lifecycle_state as state,
  time_created
from
  oci_mysql_configuration 
where
  lifecycle_state = 'DELETED'
+--------------+----+-------------+-------+--------------+
| display_name | id | description | state | time_created |
+--------------+----+-------------+-------+--------------+
+--------------+----+-------------+-------+--------------+
> 
```
</details>

### CODE REVIEW

File: oci/service.go [hunk 1]
```diff
@@ -1081,6 +1082,44 @@ func mySQLBackupService(ctx context.Context, d *plugin.QueryData, region string)
 	return sess, nil
 }
 
+// mySQLBackupService returns the service client for OCI MySQL Backup Service
+func mySQLaasService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
```
Comment (bigdatasourav) [inline oci/service.go:1086]: Fix the comment and cache key

File: oci/table_oci_mysql_configuration.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+			{
+				Name:        "time_created",
+				Description: "Time that budget was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+
+			// other columns
+			{
+				Name:        "parent_configuration_id",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:60]: parent_configuration_id should be in top columns and description in other columns

File: oci-test/tests/oci_mysql_configuration/variables.tf [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  description = "Oracle My SQL Shape."
+}
+
+provider "oci" {
+  tenancy_ocid        = var.tenancy_ocid
+  config_file_profile = var.config_file_profile
+}
+
+locals {
+  path         = "${path.cwd}/output.json"
```
Comment (bigdatasourav) [inline oci-test/tests/oci_mysql_configuration/variables.tf:43]: Fix this space issue

File: oci/service.go [hunk 4]
```diff
@@ -60,6 +60,7 @@ type session struct {
 	KmsVaultClient                 keymanagement.KmsVaultClient
 	LoggingManagementClient        logging.LoggingManagementClient
 	MonitoringClient               monitoring.MonitoringClient
+	MySQLaasClient                 mysql.MysqlaasClient
```
Comment (bigdatasourav) [inline oci/service.go:63]:
```suggestion
	MySQLConfigurationClient                 mysql.MysqlaasClient
```

File: oci/service.go (hunk 1 above)
Comment (bigdatasourav) [inline oci/service.go:1086]: can we name the service mySQLConfigurationService

File: oci/table_oci_mysql_configuration.go [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+			},
+			{
+				Name:        "time_updated",
+				Description: "The date and time the Configuration was last updated.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeUpdated.Time"),
+			},
+			{
+				Name:        "type",
+				Description: "The Configuration type, DEFAULT or CUSTOM..",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:79]:
```suggestion
				Description: "The Configuration type, DEFAULT or CUSTOM.",
```

File: oci/table_oci_mysql_configuration.go [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the Configuration.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "description",
+				Description: "The current state of the Configuration",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:43]:
```suggestion
				Description: "The current state of the Configuration.",
```

File: oci/table_oci_mysql_configuration.go [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the Configuration",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:48]:
```suggestion
				Description: "The current state of the Configuration.",
```

File: oci/table_oci_mysql_configuration.go [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+				Transform:   transform.FromField("TimeUpdated.Time"),
+			},
+			{
+				Name:        "type",
+				Description: "The Configuration type, DEFAULT or CUSTOM..",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "variables",
+				Description: "ConfigurationVariables User controllable service variables.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:84]:
```suggestion
				Description: "User controllable service variables of the Configuration.",
```

File: oci/table_oci_mysql_configuration.go [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the Configuration",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "Time that budget was created.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:53]: correct the description.

File: oci/table_oci_mysql_configuration.go [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+	logger.Debug("getConfiguration", "Compartment", compartment, "OCI_REGION", region)
+
+	var id string
+	if h.Item != nil {
+		configuration := h.Item.(mysql.ConfigurationSummary)
+		id = *configuration.Id
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+		// handle empty id in get call
+		if strings.TrimSpace(id) == "" {
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:194]: we can have this empty check outside if else block

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@rajeshbal65 suggested some changes.

File: oci/service.go [hunk 11]
```diff
@@ -1081,6 +1082,44 @@ func mySQLBackupService(ctx context.Context, d *plugin.QueryData, region string)
 	return sess, nil
 }
 
+// mySQLConfigurationService returns the service client for OCI MySQL aac Service
```
Comment (bigdatasourav) [inline oci/service.go:1085]:
```suggestion
// mySQLConfigurationService returns the service client for OCI MySQL Configuration Service
```

File: oci/service.go [hunk 12]
```diff
@@ -1081,6 +1082,44 @@ func mySQLBackupService(ctx context.Context, d *plugin.QueryData, region string)
...
+func mySQLConfigurationService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
+	logger := plugin.Logger(ctx)
+	serviceCacheKey := fmt.Sprintf("mySQLaas-%s", region)
```
Comment (bigdatasourav) [inline oci/service.go:1088]:
```suggestion
	serviceCacheKey := fmt.Sprintf("mysqlconfiguration-%s", region)
```

File: oci/table_oci_mysql_configuration.go [hunk 13]
```diff
@@ -0,0 +1,265 @@
...
+
+//// TABLE DEFINITION
+
+func tableMySQLConfiguration(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_mysql_configuration",
+		Description: "OCI My SQL Configuration",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getConfiguration,
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:23]:
```suggestion
			Hydrate:    getMySQLConfiguration,
```

File: oci/table_oci_mysql_configuration.go [hunk 14]
```diff
@@ -0,0 +1,265 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return response.Configuration, nil
+}
+
+//// TRANSFORM FUNCTION
+
+func configurationTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:227]:
```suggestion
func mySQLConfigurationTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```

File: oci/table_oci_mysql_configuration.go [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+			},
+
+			// OCI standard columns
+			{
+				Name:        "compartment_id",
+				Description: ColumnDescriptionCompartment,
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration.go:118]: compartment ID will be tenant ID in this table.

File: docs/tables/oci_mysql_configuration.md [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+  display_name,
+  id,
+  description,
+  lifecycle_state as state,
+  time_created
+from
+  oci_mysql_configuration;
+```
+
+### List deleted configurations
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_configuration.md:20]: remove this example, as for standard ones you will not get deleted.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**